alg/teca_model_segmentation.py

```
@@ -0,0 +1,196 @@
+        """
+        set Pytorch pretrained model
+        """
+        self.model = model
+        self.model.eval()
+
+    def get_report_callback(self):
+        """
+        return a teca_algorithm::report function adding the output name
+        that will hold the output predictions of the used model.
+        """
+        def report(port, rep_in):
+            rep_temp = rep_in[0]
+
+            rep = teca_py.teca_metadata(rep_temp)
+
+            if not rep.has('variables'):
+                print("variables key doesn't exist")
+                rep['variables'] = teca_py.teca_variant_array.New(np.array([]))
+
+            if self.pred_name:
+                rep.append("variables", self.pred_name)
+
+            return rep
+        return report
+
+    def get_request_callback(self):
+        """
+        return a teca_algorithm::request function adding the variable name
+        that the pretrained model will process.
+        """
+        def request(port, md_in, req_in):
+            if not self.variable_name:
+                if rank == 0:
+                    raise ValueError("ERROR: No variable to request specifed")
+
+            req = teca_py.teca_metadata(req_in)
+
+            arrays = []
+            if req.has('arrays'):
+                arrays = req['arrays']
+
+            arrays.append(self.variable_name)
+            req['arrays'] = arrays
+
+            return [req]
+        return request
+
+    def get_execute_callback(self):
+        """
+        return a teca_algorithm::execute function
+        """
+        def execute(port, data_in, req):
+            """
+            expects an array of an input variable to run through
+            the torch model and get the segmentation results as an
+            output.
+            """
+            in_mesh = teca_py.as_teca_cartesian_mesh(data_in[0])
+
+            if in_mesh is None:
+                if rank == 0:
+                    raise ValueError("ERROR: empty input, or not a mesh")
```

could we raise the exception on all ranks?